### PR TITLE
Mustache Template Update

### PR DESCRIPTION
**Pull request by thomas-huston-zocdoc**
@ZocDoc/polaris-devs @Dillen-Roggensinger-ZocDoc 

Changes
--
- The way we currently compile our mustache templates explicitly attaches them to `window.JST`. That's going to cause issues when we try to namespace all our JS code for syndication. This updates the way we format our compiled mustache templates so that they attach to `this`, not `window`. While I was there, I also made them `require`/`module.exports` friendly.

### CODE REVIEW

File: src/compile/compile-mustache.js [hunk 1]
```diff
@@ -11,32 +13,44 @@ var Promise = require('bluebird');
...
 
-            var compiledTemplate = compileTemplate(options.code);
+            return {
+                code: result
+            };
 
-            var templateObject = '{ code: ' + compiledTemplate + ', partials: {}, subs: {} }';
+        });
 
-            var result = 'window["JST"] = window["JST"] || {};'
```
Comment (thomas-huston-zocdoc) [inline src/compile/compile-mustache.js]: Once this started to get more complex, concatenating strings of code just got super annoying. I'm now using Handlebars and a separate template file to generate the output.

File: src/compile/mustache.hbs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+(function() {
```
Comment (thomas-huston-zocdoc) [inline src/compile/mustache.hbs]: This is the new format of our compiled mustache templates.

File: tests/integration/helpers/integration-test-helper.js [hunk 3]
```diff
@@ -136,7 +136,7 @@ TestUtility.prototype.VerifyFileContents = function (dir, fileName, expectedCont
     var _this = this;
     _this.runFunc(function () {
         var bundleContents = _this.FileSystem.readFileSync(dir + '/' + fileName, 'utf8');
-        expect(bundleContents).toBe(expectedContents);
+        expect(bundleContents.replace(/\s/g, '')).toBe(expectedContents.replace(/\s/g, ''));
```
Comment (thomas-huston-zocdoc) [inline tests/integration/helpers/integration-test-helper.js]: @Andrew-Hagedorn-ZocDoc This makes our integration tests ignore whitespace when comparing text. I got fed up with trying to make sure it all matched perfectly, it was really hard to find missing spaces in the jasmine test error messages.

File: tests/integration/folder-option-spec.js [hunk 4]
```diff
@@ -62,7 +62,7 @@ test.describeIntegrationTest("Outputting to another directory:", function() {
             test.assert.verifyBundleIs(
                 ';var file1="file1";\n' +
                 ';var file2="file2";\n' +
-                ';window.JST=window.JST||{},JST.file3=new Hogan.Template({code:function(i,n,e){var a=this;return a.b(e=e||""),a.b("<div> "),a.b(a.v(a.f("c",i,n,0))),a.b(" </div>"),a.fl()},partials:{},subs:{}});\n'
+                ';(function(){vare,i,t="undefined"!=typeofmodule&&module.exports;e=t?require("hogan"):this.Hogan,i=newe.Template({code:function(e,i,t){varo=this;returno.b(t=t||""),o.b("<div>"),o.b(o.v(o.f("c",e,i,0))),o.b("</div>"),o.fl()},partials:{},subs:{}}),t?module.exports=i:(this.JST=this.JST||{},JST.file3=i)}).call(this);\n'
```
Comment (Dillen-Roggensinger-ZocDoc) [inline tests/integration/folder-option-spec.js]: I don't know much about what this is but `i=newe.Template` seems wrong. Shouldn't it be `i=new e.Template`?

Comment (thomas-huston-zocdoc) [inline tests/integration/folder-option-spec.js]: This is just me copy-pasting code with whitespace removed. I'll update it so it doesn't look so dumb.

File: tests/integration/basic-js-spec.js [hunk 5]
```diff
@@ -139,7 +139,7 @@ test.describeIntegrationTest("Js Bundling:", function() {
         test.actions.Bundle();
 
         test.assert.verifyBundleIs(
-            ';window.JST=window.JST||{},JST.file1=new Hogan.Template({code:function(i,n,a){var e=this;return e.b(a=a||""),e.b("<div> "),e.b(e.v(e.f("a",i,n,0))),e.b(" </div>"),e.fl()},partials:{},subs:{}});\n' +
+            ';(function(){var e,i,t="undefined"!=typeof module&&module.exports;e=t?require("hogan"):this.Hogan,i=new e.Template({code:function(e,i,t){var o=this;return o.b(t=t||""),o.b("<div> "),o.b(o.v(o.f("a",e,i,0))),o.b(" </div>"),o.fl()},partials:{},subs:{}}),t?module.exports=i:(this.JST=this.JST||{},JST.file1=i)}).call(this);\n' +
```
Comment (thomas-huston-zocdoc) [inline tests/integration/basic-js-spec.js]: @Dillen-Roggensinger-ZocDoc Fixed these tests so they don't look dumb. Good catch.

File: tests/integration/helpers/integration-test-helper.js [hunk 6]
```diff
@@ -136,6 +136,16 @@ TestUtility.prototype.VerifyFileContents = function (dir, fileName, expectedCont
     var _this = this;
     _this.runFunc(function () {
         var bundleContents = _this.FileSystem.readFileSync(dir + '/' + fileName, 'utf8');
+        if (expectedContents.length !== bundleContents.length) {
```
Comment (thomas-huston-zocdoc) [inline tests/integration/helpers/integration-test-helper.js]: @Andrew-Hagedorn-ZocDoc Removing the whitespace before comparing ended up being a little weird. Instead, I've added some console logging that helped me pinpoint exactly what the mismatch in the string was. Turns out it was the newline characters.

**Pull request merged**